dbt/adapters/default.py

```diff
@@ -539,7 +540,19 @@ def execute_and_fetch(cls, profile, sql, model_name=None,
                           auto_begin=False):
         _, cursor = cls.execute_one(profile, sql, model_name, auto_begin)
 
-        return cursor.fetchall()
+        status = cls.get_status(cursor)
+        rows = cursor.fetchall()
+        return status, rows
+
+    @classmethod
+    def execute(cls, profile, sql, model_name=None, auto_begin=False,
+                fetch=False):
+        if fetch:
+            return cls.execute_and_fetch(profile, sql, model_name, auto_begin)
+        else:
+            _, cursor = cls.execute_one(profile, sql, model_name, auto_begin)
+            status = cls.get_status(cursor)
+            return status, None
```

Comment:
default result data was `[]`, now it's `None`

I really like this approach but I think we should condense the `execute` / `execute_one` / `execute_and_fetch` / `add_query` APIs into a smaller set of functions. Not blocking for this PR.